### PR TITLE
[RFR][TA]Nodejs Provider Analysis

### PR DESCRIPTION
**Pull request by kpunwatk**
Resolves https://issues.redhat.com/browse/MTA-3372
[Jenkins run](https://main-jenkins-csb-migrationqe.apps.ocp-c1.prod.psi.redhat.com/view/MTA/job/mta/job/mta-cli-tests-runner/142/console) 
Screenshot from 2024-09-13 14-07-16

### CODE REVIEW

**Comment by kpunwatk** [comment]
@abrugaro @sshveta Could you please review this PR? Thanks!

File: data/yaml/python_rules.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+- message: python sample rule 001
```
Comment (abrugaro) [inline data/yaml/python_rules.yaml:2]: Why are we adding a python rule in the node js provider test PR ?

File: tests/test_selective_nodejs_provider.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import os
+import subprocess
+
+from utils import constants
+from utils.command import build_analysis_command
+from utils.report import get_json_from_report_output_file
+
+
+# Polarion TC MTA-542
+def test_nodejs_provider_analysis_with_app(nodejs_analysis_data):
```
Comment (abrugaro) [inline tests/test_selective_nodejs_provider.py:10]: Can you open a different PR to rename this method and the `go` to remove the `with_app`? it is a little redundant.

`test_nodejs_provider_analysis`

**Comment by abrugaro** [comment]
Moved to https://github.com/konveyor-ecosystem/kantra-cli-tests/pull/32

**Pull request closed without merging**